src/zc/buildout/easy_install.py

```diff
@@ -213,20 +267,51 @@ def __init__(self,
 
         self._index_url = index
         path = (path and path[:] or []) + buildout_and_setuptools_path
-        if dest is not None and dest not in path:
-            path.insert(0, dest)
         self._path = path
         if self._dest is None:
             newest = False
         self._newest = newest
-        self._env = pkg_resources.Environment(path)
+        self._env = self._make_env()
         self._index = _get_index(index, links, self._allow_hosts)
         self._requirements_and_constraints = []
         self._check_picked = check_picked
 
         if versions is not None:
             self._versions = normalize_versions(versions)
 
+    def _make_env(self):
+        full_path = self._get_dest_dist_paths() + self._path
+        env = pkg_resources.Environment(full_path)
+        # this needs to be called whenever self._env is modified (or we could
+        # make an Environment subclass):
+        self._eggify_env_dest_dists(env, self._dest)
+        return env
+
+    def _env_rescan_dest(self):
```

Comment:
> Only used once. Please inline.

I know you mentioned you don't think it's necessary, but `_env_rescan_dest()` and  `_make_env()` are the only methods that modify `self._env`.

I feel that if we just inline it we might forget to `_eggify_env_dest_dists()` in the future if we add another place that modifies `self._env`.

Or we could use a subclass of `Environment`. We would only need to override the `Environment.add()` method in this case.

But it's your call.